react-front-end/tsrc/search/components/GallerySearchResult.tsx

```diff
@@ -40,25 +49,84 @@ const GallerySearchResult = ({ items }: GallerySearchResultProps) => {
     LightboxProps | undefined
   >();
 
-  // A list of LightboxEntry which includes all main entries and additional entries.
-  const lightboxEntries: LightboxEntry[] = items.flatMap(
-    ({ mainEntry, additionalEntries }) =>
-      [mainEntry, ...additionalEntries].map(
-        ({ id, name, mimeType, directUrl }) => ({
-          src: directUrl,
-          title: name,
-          mimeType: mimeType,
-          id,
-        })
-      )
-  );
+  const [galleryItems, setGalleryItems] =
+    useState<GallerySearchResultItem[]>(items);
+
+  // Function to update the Item list, rebuild the Lightbox entry list and return a lightboxHandler.
+  const updateGalleryItemList = (
+    newItem: GallerySearchResultItem
+  ): LightboxHandler => {
+    const newItems = galleryItems.map((i) =>
+      i.uuid === newItem.uuid && i.version === newItem.version ? newItem : i
+    );
```

Comment:
I wonder whether it's worth adding a comment such as:

```suggestion
    // Update replace existing gallery item with newItem
    const newItems = galleryItems.map((i) =>
      i.uuid === newItem.uuid && i.version === newItem.version ? newItem : i
    );
```

Or, would it be better to just have a function like

```typescript
const updateGalleryItem = (item: GallerySearchResultItem): GallerySearchResultItem[] => {
  const updatedItems = galleryItems.map((i) =>
    i.uuid === newItem.uuid && i.version === newItem.version ? newItem : i
  );
  setGalleryItems(updatedItems);

  return updatedItems;
}
```

Reply:
I like renaming this const to `updatedItems`.